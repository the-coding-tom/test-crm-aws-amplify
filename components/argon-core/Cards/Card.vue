<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
