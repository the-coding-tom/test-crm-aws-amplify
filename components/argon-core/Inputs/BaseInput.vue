<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
