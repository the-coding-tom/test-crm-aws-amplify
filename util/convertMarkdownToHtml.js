export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
