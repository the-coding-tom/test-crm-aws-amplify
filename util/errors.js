export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
