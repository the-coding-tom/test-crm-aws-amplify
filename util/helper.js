const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
