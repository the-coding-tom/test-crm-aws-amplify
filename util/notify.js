const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
