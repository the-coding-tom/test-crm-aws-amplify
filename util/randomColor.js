const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
