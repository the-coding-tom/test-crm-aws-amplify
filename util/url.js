export const getQueryParams = url => {
  const urlParam = require('url')

  const parsedUrl = urlParam.parse(url)

  return parsedUrl.search
}
