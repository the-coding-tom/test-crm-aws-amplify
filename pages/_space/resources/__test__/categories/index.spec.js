import Card from '~/components/argon-core/Cards/Card.vue'
import AddPage from '@/pages/_space/resources/categories/add.vue'
import IndexPage from '@/pages/_space/resources/categories/index.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BootstrapVue from 'bootstrap-vue'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'

import allCategories from '../resources/all-categories.json'

describe('Page: /:space/resources/categories | Categories Index Page', () => {
  let NuxtStore, store, wrapper, $resource, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $resource = {
      getAllCategories: jest
        .fn()
        .mockResolvedValueOnce({ data: [] })
        .mockResolvedValue(cloneDeep(allCategories)),
      updateCategory: jest.fn().mockResolvedValue(),
      deleteCategory: jest.fn().mockResolvedValue()
    }

    // mock confirm for removal
    global.confirm = jest
      .fn()
      .mockReturnValueOnce(true)
      .mockReturnValueOnce(false)
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.$resource = $resource

    // mock toast
    store._vm.$bvToast = {
      toast: jest.fn()
    }

    router = new VueRouter({
      routes: [
        {
          name: 'space-resources-categories-add',
          path: '_space/resources/categories/add',
          component: AddPage
        }
      ]
    })

    wrapper = createWrapper(IndexPage, store, router, {}, {}, vueInstance => {
      vueInstance.use(BootstrapVue)
      vueInstance.component(BaseInput.name, BaseInput)
      vueInstance.component(Card.name, Card)
      vueInstance.prototype.$resource = $resource
    })

    await wrapper.vm.$options.asyncData({ store })
  })

  it('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('get all categories', async () => {
    const wrapArr = wrapper.findAll('label.custom-control-label')

    expect(wrapArr.length).toBe(2)
    expect(wrapArr.at(0).text()).toBe(allCategories.data.data[0].name)
    expect(wrapArr.at(1).text()).toBe(allCategories.data.data[1].name)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('update a category', async () => {
    // set the first radio input to active
    wrapper.find('input.custom-control-input').setChecked(true)
    await flushPromises()

    // check if input has been updated with values
    expect(wrapper.find('#name').element.value).toBe(
      allCategories.data.data[0].name
    )
    expect(wrapper.find('#color').element.value).toBe(
      allCategories.data.data[0].color
    )

    // change it value to 'get-together'
    wrapper.find('#name').setValue('get-together')

    // click on update
    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($resource.updateCategory).toHaveBeenCalled()
    expect($resource.getAllCategories).toHaveBeenCalled()
  })

  it('remove a category', async () => {
    // set the first radio input to active
    wrapper.find('input.custom-control-input').setChecked(true)
    await flushPromises()

    // click remove on remove category button
    wrapper
      .find('button.btn.text-danger.mr-r-20.btn-transparent')
      .trigger('click')
    await flushPromises()

    expect($resource.getAllCategories).toHaveBeenCalled()
    expect($resource.deleteCategory).toHaveBeenCalled()
  })
})
