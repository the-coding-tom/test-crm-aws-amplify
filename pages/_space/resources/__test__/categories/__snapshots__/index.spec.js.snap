// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/resources/categories | Categories Index Page get all categories 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between py-4">
            <h1 class="page-title"><span>Resource Categories</span> <b style="display: none;">•</b> </h1>
            <div><button type="button" disabled="disabled" class="btn text-danger mr-r-20 btn-transparent disabled"><i class="fa fa-times"></i> Remove Category</button> <button type="submit" class="btn btn btn-primary btn-secondary" style="color: white;">Update Category</button></div>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6">
      <div class="card-deck flex-column flex-xl-row">
        <div class="card">
          <!---->
          <div class="card-body">
            <h3 class="mr-b-20">Update Category</h3>
            <div class="row">
              <div class="form-group col-md-4"><label class="form-control-label">
                  Name
                </label>
                <div class="input-group has-label">
                  <!----> <input type="text" valid="true" required="required" id="name" name="name" placeholder="Category Name" class="form-control">
                  <!---->
                  <!---->
                  <!---->
                </div>
              </div>
              <div class="form-group col-md-3"><label class="form-control-label">
                  Color Tag
                </label>
                <div class="input-group has-label">
                  <!----> <input type="text" valid="true" required="required" id="color" placeholder="#4C4D4F" class="form-control">
                  <!---->
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <div class="row mr-t-40">
              <div role="radiogroup" tabindex="-1" class="col-md-6 bv-no-focus-ring" id="__BVID__21">
                <div class="row">
                  <div class="col-md-6">
                    <div class="custom-control custom-control-inline custom-radio"><input type="radio" name="__BVID__21" autocomplete="off" class="custom-control-input" value="a2cc76b7-9907-4d86-8072-c1b9ef276a0a" id="__BVID__22"><label class="custom-control-label" for="__BVID__22">conference</label></div>
                  </div>
                  <div class="col-md-6">
                    <div class="custom-control custom-control-inline custom-radio"><input type="radio" name="__BVID__21" autocomplete="off" class="custom-control-input" value="aa9c7423-4022-491f-ad3e-401eec72b8d8" id="__BVID__23"><label class="custom-control-label" for="__BVID__23">meeting</label></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Page: /:space/resources/categories | Categories Index Page renders properly 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between py-4">
            <h1 class="page-title"><span>Resource Categories</span> <b style="display: none;">•</b> </h1>
            <div><button type="button" disabled="disabled" class="btn text-danger mr-r-20 btn-transparent disabled"><i class="fa fa-times"></i> Remove Category</button> <button type="submit" class="btn btn btn-primary btn-secondary" style="color: white;">Update Category</button></div>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6">
      <div class="card-deck flex-column flex-xl-row">
        <div class="card">
          <!---->
          <div class="card-body">
            <h3 class="mr-b-20">Update Category</h3>
            <div class="row">
              <div class="form-group col-md-4"><label class="form-control-label">
                  Name
                </label>
                <div class="input-group has-label">
                  <!----> <input type="text" valid="true" required="required" id="name" name="name" placeholder="Category Name" class="form-control">
                  <!---->
                  <!---->
                  <!---->
                </div>
              </div>
              <div class="form-group col-md-3"><label class="form-control-label">
                  Color Tag
                </label>
                <div class="input-group has-label">
                  <!----> <input type="text" valid="true" required="required" id="color" placeholder="#4C4D4F" class="form-control">
                  <!---->
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <div class="row mr-t-40">
              <div role="radiogroup" tabindex="-1" class="col-md-6 bv-no-focus-ring" id="__BVID__11">
                <div class="row"></div>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </form>
</div>
`;
